compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -307,6 +308,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 // a reference type wherefore peeling doesn't give up any expressiveness.
                 _ => AdjustMode::Peel,
             },
+            // A never pattern behaves somewhat like a literal or unit variant.
```

Comment:
The peel semantics here mean that `match &! { ! => .. }` works, right? Is that intended?

Reply:
That is intended. That's one of the unresolved questions for the feature